Makefile

```diff
@@ -33,7 +22,10 @@ IMAGE_TAG ?= latest
 ORIGINAL_IMAGE_TAG ?=
 STRUCTURE_TEST_TARGET ?= normal
 
-ENVTEST_KUBERNETES_VERSION=1.27
+CONTROLLER_RUNTIME_VERSION := $(shell awk '/sigs\.k8s\.io\/controller-runtime/ {print substr($$2, 2)}' go.mod)
+CONTROLLER_TOOLS_VERSION := $(shell awk '/sigs\.k8s\.io\/controller-tools/ {print substr($$2, 2)}' go.mod)
+
+ENVTEST_KUBERNETES_VERSION := $(shell echo $(KUBERNETES_VERSION) | cut -d "." -f 1-2)
```

Comment:
These (and `PROTOC_GEN_GO_VERSION`, ` PROTOC_GEN_DOC_VERSION`,  
`PROTOC_GEN_GO_GRPC_VERSION`) are not defined in `version.mk`? I understand these versions are special in the sense that these are actually defined in `go.mod` rather than `Makefile`, but I think it would be easy to understand if `versions.mk` contains all version numbers of our dependencies that is referenced in `Makefile`s, instead of only part of these.

Reply:
agreed. I'll revise it.